tests/modules/cmdintrospection.c

```diff
@@ -0,0 +1,171 @@
+            {"7.0.0", "Added support for the `<ms>-*` explicit ID form."},
+            {0}
+        },
+        .key_specs = (RedisModuleCommandKeySpec[]){
+            (RedisModuleCommandKeySpec){
+                .flags = REDISMODULE_CMD_KEY_WRITE,
+                .begin_search_type = REDISMODULE_KSPEC_BS_INDEX,
+                .bs.index.pos = 1,
+                .find_keys_type = REDISMODULE_KSPEC_FK_RANGE,
+                .fk.range = {0,1,0}
+            },
+            {0}
+        },
+        .args = (RedisModuleCommandArg[]){
+            (RedisModuleCommandArg){
+                .name = "key",
+                .type = REDISMODULE_ARG_TYPE_KEY,
+                .key_spec_index = 0
+            },
+            (RedisModuleCommandArg){
+                .name = "nomkstream",
+                .type = REDISMODULE_ARG_TYPE_PURE_TOKEN,
+                .key_spec_index = -1,
+                .token = "NOMKSTREAM",
+                .since = "6.2.0",
+                .flags = REDISMODULE_CMD_ARG_OPTIONAL
+            },
+            (RedisModuleCommandArg){
+                .name = "trim",
+                .type = REDISMODULE_ARG_TYPE_BLOCK,
+                .key_spec_index = -1,
+                .flags = REDISMODULE_CMD_ARG_OPTIONAL,
+                .subargs = (RedisModuleCommandArg[]){
+                    (RedisModuleCommandArg){
+                        .name = "strategy",
+                        .type = REDISMODULE_ARG_TYPE_ONEOF,
+                        .key_spec_index = -1,
+                        .subargs = (RedisModuleCommandArg[]){
+                            (RedisModuleCommandArg){
+                                .name = "maxlen",
+                                .type = REDISMODULE_ARG_TYPE_PURE_TOKEN,
+                                .key_spec_index = -1,
+                                .token = "MAXLEN",
+                            },
+                            (RedisModuleCommandArg){
+                                .name = "minid",
+                                .type = REDISMODULE_ARG_TYPE_PURE_TOKEN,
+                                .key_spec_index = -1,
+                                .token = "MINID",
+                                .since = "6.2.0",
+                            },
+                            {0}
+                        }
+                    },
+                    (RedisModuleCommandArg){
+                        .name = "operator",
+                        .type = REDISMODULE_ARG_TYPE_ONEOF,
+                        .key_spec_index = -1,
+                        .flags = REDISMODULE_CMD_ARG_OPTIONAL,
+                        .subargs = (RedisModuleCommandArg[]){
+                            (RedisModuleCommandArg){
+                                .name = "equal",
+                                .type = REDISMODULE_ARG_TYPE_PURE_TOKEN,
+                                .key_spec_index = -1,
+                                .token = "="
+                            },
```

Comment:
I suggest we allow `key_spec_index` to be 0 (and still interpret it as -1 when type isn't KEY). This will allow it to be left out in compound literals such as this one. All the other fields are zero or NULL by default so it's fine to leave them out already. Only this one needs to be set to -1 in a lot of places.

Reply:
Following my own suggestion, I added commit "Allow .key_spec_index = -1 to be omitted". I'll revert it if it's not acceptable.